src/pages/cart/index.tsx

```diff
@@ -1,5 +1,42 @@
-import { ReactElement } from 'react';
+import { ReactElement, useContext } from 'react';
+import { CartItem } from '../../components/general/cart-item';
+import { CartContext } from '../../contexts/Cart';
 
 export const CartPage = (): ReactElement => {
-  return <div>Cart Page</div>;
+  const { cart, setCart } = useContext(CartContext);
+  type Amount = 'increment' | 'decrement';
+  const changeQuantity = (id: string, amount: Amount) => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          if (amount === 'increment')
+            return { ...currItem, quantity: currItem.quantity + 1 };
+          if (amount === 'decrement')
+            return { ...currItem, quantity: currItem.quantity - 1 };
```

Comment:
For the decrement functionality, should it stop at 0 so you can't have negative items or maybe remove the item when the user hits 0?

Reply:
good point to stop it as 0. instead of having 'increment' or 'decrement', you could just pass a number as the amount and do something like { ...currItem, quantity: currItem.quantity + amount };

it also means that in the future you can add like 3 using this same function at once by using a select in the product details page

Reply:
All Done!